src/sign/mod.rs

```diff
@@ -1,10 +1,387 @@
 //! DNSSEC signing.
 //!
 //! **This module is experimental and likely to change significantly.**
-#![cfg(feature = "sign")]
-#![cfg_attr(docsrs, doc(cfg(feature = "sign")))]
+//!
+//! Signatures are at the heart of DNSSEC -- they confirm the authenticity of
+//! a DNS record served by a security-aware name server.  Signatures can be
+//! made "online" (in an authoritative name server while it is running) or
+//! "offline" (outside of a name server).  Once generated, signatures can be
+//! serialized as DNS records and stored alongside the authenticated records.
+//!
+//! A DNSSEC key actually has two components: a cryptographic key, which can
+//! be used to make and verify signatures, and key metadata, which defines how
+//! the key should be used.  These components are brought together by the
+//! [`SigningKey`] type.  It must be instantiated with a cryptographic key
+//! type, such as [`common::KeyPair`], in order to be used.
+//!
+//! # Example Usage
+//!
+//! At the moment, only "low-level" signing is supported.
```

Comment:
Rather than making users copy that example whenever they want to perform an import, we should just expose an `std`-gated function on `SigningKey` to do it for them (given the file name(s) to lead from).  What do you think?

Reply:
I'm not sure what you are suggesting. If you are suggesting an fn to load from a file on disk I would say no don't do that as we had said we were not going to put file based APIs into domain.

Reply:
I think that's unfortunate, but alright.  I'll add an example here.